src/Microsoft.Extensions.OData.Migration/Formatters/Serialization/ODataMigrationOutputFormatter.cs

```diff
@@ -0,0 +1,385 @@
+            writerSettings.BaseUri = baseAddress;
+            writerSettings.Version = ODataVersion.V4; // Todo how to specify v3?  Maybe don't because reading as v4
+            writerSettings.Validations = writerSettings.Validations & ~ValidationKinds.ThrowOnUndeclaredPropertyForNonOpenType;
+
+            string metadataLink = internalUrlHelper.CreateODataLink(MetadataSegment.Instance);
+
+            if (metadataLink == null)
+            {
+                throw new SerializationException("Unable to determine metadata url");
+            }
+
+            writerSettings.ODataUri = new ODataUri
+            {
+                ServiceRoot = baseAddress,
+
+                // TODO: 1604 Convert webapi.odata's ODataPath to ODL's ODataPath, or use ODL's ODataPath.
+                SelectAndExpand = internalRequest.ODataFeature()?.SelectExpandClause,
+                Apply = internalRequest.ODataFeature().ApplyClause,
+                Path = (path == null || IsOperationPath(path)) ? null : path.Path,
+            };
+
+            ODataMetadataLevel metadataLevel = ODataMetadataLevel.MinimalMetadata;
+            if (contentType != null)
+            {
+                IEnumerable<KeyValuePair<string, string>> parameters =
+                    contentType.Parameters.Select(val => new KeyValuePair<string, string>(val.Name.Value, val.Value.Value));
+                metadataLevel = ODataMediaTypes.GetMetadataLevel(contentType.MediaType.Value, parameters);
+            }
+
+            using (ODataMessageWriter messageWriter = new ODataMessageWriter(responseMessage, writerSettings, model))
+            {
+                ODataSerializerContext writeContext = getODataSerializerContext();
+                writeContext.NavigationSource = targetNavigationSource;
+                writeContext.Model = model;
+                writeContext.RootElementName = GetRootElementName(path) ?? "root";
+                writeContext.SkipExpensiveAvailabilityChecks = serializer.ODataPayloadKind == ODataPayloadKind.ResourceSet;
+                writeContext.Path = path;
+                writeContext.SelectExpandClause = internalRequest.ODataFeature()?.SelectExpandClause;
+                writeContext.MetadataLevel = metadataLevel;
+
+                // Substitute stream to swap @odata.context
+                Stream substituteStream = new MemoryStream();
+                Stream originalStream = messageWriter.SubstituteResponseStream(substituteStream);
+
+                serializer.WriteObject(value, type, messageWriter, writeContext);
+                StreamReader reader = new StreamReader(substituteStream);
+                substituteStream.Seek(0, SeekOrigin.Begin);
+                JToken responsePayload = JToken.Parse(reader.ReadToEnd());
+
+                // If odata context is present, replace with odata metadata
+                if (responsePayload["@odata.context"] != null)
```

Comment:
These standard OData names should be defined as string constants and not hard-coded everywhere.